go-controller/pkg/ovn/egressip.go

```diff
@@ -742,18 +853,22 @@ func (e *EgressIPController) deleteEgressIPAssignments(name string, statusesToRe
 				// we can continue here since this pod was not managed by this statusToRemove
 				continue
 			}
+			podNamespace, podName := getPodNamespaceAndNameFromKey(podKey)
+			ni, err := e.nadController.GetActiveNetworkForNamespace(podNamespace)
```

Comment:
this may not work if the namespace is gone. You may want to store the active network in the pod assignment cache at add time.

Reply:
is the network is gone, then all the OVN constructs to support EgressIP are removed. Ill add a unit test cases for this.

Reply:
I added a commit on top to address this.